merkletree/proof.go

```diff
@@ -65,7 +64,7 @@ func authPathHash(ap *AuthenticationPath,
 
 // VerifyAuthPath should be called after the vrf index is verified successfully
 func VerifyAuthPath(ap *AuthenticationPath,
-	leafIndex, leafCommitment []byte, leafLevel uint32, isLeafEmpty bool,
+	leafIndex []byte, leafCommitment *crypto.Commit, leafLevel uint32, isLeafEmpty bool,
```

Comment:
> And in the caller we could pass `leaf.Commitment().Value`

Ah, right, so the reason I did that was because `leaf.Commitment()` is `nil` for empty nodes, so you get a nil pointer dereference.